R/Model-methods.R

```diff
@@ -116,114 +116,122 @@ setMethod(
-                                  "dose")
-              ## now call the function with dose and with
-              ## the arguments taken from the samples
-              ret <- do.call(probFun,
-                             c(list(dose=dose),
-                               samples@data[argNames]))
-              ## return the resulting vector
-              return(ret)
-            })
+#' @rdname prob
+#'
+#' @description Compute toxicity probabilities for a given dose, a given single
+#'   agent dose escalation model, and the samples.
+#'
+#' @aliases prob-Model
+#' @example examples/Model-method-prob.R
+#' @export
+#'
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "Model",
+    samples = "Samples"
+  ),
+  definition = function(dose, model, samples, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
+    prob_args <- setdiff(formalArgs(prob_fun), "dose")
+    do.call(prob_fun, c(list(dose = dose), samples@data[prob_args]))
+  }
+)
 
+# prob-ModelTox ----
 
+#' @rdname prob
+#'
+#' @description Compute toxicity probabilities for a given dose, a given Pseudo
+#'   DLE (dose-limiting events)/toxicity model, and the samples.
+#'
+#' @aliases prob-ModelTox
+#' @example examples/Model-method-prob-ModelTox.R
+#' @export
+#'
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "ModelTox",
+    samples = "Samples"
+  ),
+  definition = function(dose, model, samples, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
+    prob_args <- setdiff(formalArgs(prob_fun), "dose")
+    do.call(prob_fun, c(list(dose = dose), samples@data[prob_args]))
+  }
+)
 
+# prob-ModelTox_noSamples ----
 
-## ==============================================================================
-## ## Compute the probability for a given dose, given Pseudo DLE model
-## --------------------------------------------------
+#' @rdname prob
+#'
+#' @description Compute the toxicity probability for a given dose and a given
+#'   Pseudo DLE (dose-limiting events)/toxicity model without samples.
+#'   All the arguments to `model@prob` function (except `dose`) should be
+#'   present in the `model` object.
+#'
+#' @aliases prob-ModelTox_noSamples
+#' @example examples/Model-method-prob-ModelTox_noSamples.R
+#' @export
+#'
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "ModelTox",
+    samples = "missing"
+  ),
+  definition = function(dose, model, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
+    prob_args <- setdiff(formalArgs(prob_fun), "dose")
+    if (!all(prob_args %in% slotNames(model))) {
+      stop("Arguments to `model@prob` (except `dose`) must be in the `model`")
+    }
+    prob_args_values <- sapply(
+      prob_args,
+      function(a) slot(model, a),
```

Comment:
Should not need another function definition here, can just appropriately pass the model to named first argument

Reply:
Sure, thanks for the hint!